pkg/kwok/controllers/metrics.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"sigs.k8s.io/kwok/pkg/apis/internalversion"
+	kwokmetrics "sigs.k8s.io/kwok/pkg/kwok/metrics"
+	"sigs.k8s.io/kwok/pkg/kwok/metrics/cel"
+	"sigs.k8s.io/kwok/pkg/utils/maps"
+	"sigs.k8s.io/kwok/pkg/utils/slices"
+)
+
+var metricsHandler nodeMetricHandler
+
+func setMetrics(metrics []*internalversion.Metric) {
+	metricsHandler.metrics = metrics
+}
```

Comment:
Seems appropriate :relaxed:
Since PodController is a public package level variable, I wanted to still have a way of updating the node metric like total number of started containers. If someone chooses not to use Controller, but to instantiate PodController, then they would not have a way of updating metrics. In that case, the metricHandler would need to be exposed as well, and I did not want to extend the API surface with this :relaxed: 
Is that a valid concern at all? Maybe metric handler can go into `sigs.k8s.io/kwok/pkg/kwok/controllers/metrics` and have their methods public?